sc2/bot_ai.py

```diff
@@ -137,6 +137,19 @@ def expansion_locations(self) -> Dict[Point2, Units]:
         """ Returns dict with the correct expansion position Point2 key, resources (mineral field, vespene geyser) as value """
         return centers
 
+    def _correct_zerg_supply(self):
+        """ The client incorrectly rounds zerg supply down instead of up (see
+            https://github.com/Blizzard/s2client-proto/issues/123), so self.supply_used
+            and friends return the wrong value when there are an odd number of zerglings
+            and banelings. This function corrects the bad values. """
+        # TODO: remove when Blizzard/sc2client-proto#123 gets fixed.
+        correction = self.units({UnitTypeId.ZERGLING, UnitTypeId.ZERGLINGBURROWED,
+                                 UnitTypeId.BANELING, UnitTypeId.BANELINGBURROWED,
+                                 UnitTypeId.BANELINGCOCOON}).amount % 2
+        self.supply_used += correction
+        self.supply_army += correction
+        self.supply_left -= correction
```

Comment:
You dont need to do `self.supply_left -= correction if you call the function before `self.supply_left` in` _prepare_step`

Reply:
Yes, but that breaks up the nice grouping of the `supply_*` properties, making the code a little less readable. I'm not sure which I prefer.

Reply:
yeah ok... :)